functions/tx-manager_dashboard/main.py

```diff
@@ -0,0 +1,89 @@
+    global logger
+    logger.info( "starting" )
+    return generateDashboard( event, context, DynamoDBHandler, logger )
+
+def generateDashboard( event, context, DynamoDBHandler, logger ):
+    """
+    Generate page with metrics indicating configuration of tx-manager
+    :param dict event:
+    :param context:
+    :param DynamicDBHandler:
+    :param logger:
+    """
+    logger.info( "Start: generateDashboard" )
+
+    dbModule = DynamoDBHandler( "tx-module" )
+    items = dbModule.query_items()
+
+    if items and len( items ):
+        logger.info( "  Found: " + str( len( items ) ) + " item[s] in tx-module" )
+        rec = "<h1>TX-Manager Dashboard</h1><h2>Module Attributes</h2><br><table>\n"
+
+        for item in items:
+            #logger.info( json.dumps( item ) )
+            logger.info( item["name"] )
+            rec +=        '            <tr><td class="hdr" colspan="2">'                 + str( item["name"] )           + "</td></tr>\n"
+
+            # TBD the following code almosts walks the db record replacing next 11 lines
+            #for attr, val in item:
+            #    if ( attr != 'name' ) and ( len( attr ) > 0 ):
+            #      rec += '            <tr><td class="lbl" >' + attr.replace("_", " ").title() + ':</td><td>' + "lst( val )" + "</td></tr>\n"
+            #rec += '<tr><td colspan="2"></td></tr>'
+
+            rec += '            <tr><td class="lbl" >Type:</td><td>'              + str( item["type"] )           + "</td></tr>\n"
+            rec += '            <tr><td class="lbl" >Input Format:</td><td>'      + lst( item["input_format"] )   + "</td></tr>\n"
+            rec += '            <tr><td class="lbl" >Output Format:</td><td>'     + lst( item["output_format"] )  + "</td></tr>\n"
+            rec += '            <tr><td class="lbl" >Resource Types:</td><td>'    + lst( item["resource_types"] ) + "</td></tr>\n"
+            rec += '            <tr><td class="lbl" >Version:</td><td>'           + str( item["version"] )        + "</td></tr>\n"
+
+            if len( item["options"] ) > 0:
+                rec += '            <tr><td class="lbl" >Options:</td><td>'       + lst( item["options"] )        + "</td></tr>\n"
+
+            if len( item["private_links"] ) > 0:
+                rec += '            <tr><td class="lbl" >Private Links:</td><td>' + lst( item["private_links"] )  + "</td></tr>\n"
+
+            if len( item["public_links"] ) > 0:
+                rec += '            <tr><td class="lbl" >Public Links:</td><td>'  + lst( item["public_links"] )   + "</td></tr>\n"
+
+        rec += "        </table>"
+        return { 'title': 'Tx-Manager Dashboard', 'message': rec }
+    else:
+        logger.info( "No tx-modules found." )
+        return  "No tx-modules found" 
+
+def lst( listing ):
+    """
+    create comma separated list of items if argument is a list
+    :param list|string listing 
+    """
+    stringed = ""
+
+    if type( listing ) == list:
+        for item in listing:
```

Comment:
It might be simpler to use the `join(..)` method on strings here:

```
if type(listing) == list:
    return ", ".join(listing)
else:
    return listing
```